lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -530,120 +529,74 @@ internal class StaticTypeInferenceVisitorTransform(
             val nAry = super.transformExprCallAgg(node) as PartiqlAst.Expr.CallAgg
             val funcName = nAry.funcName
             // unwrap the type if this is a collectionType
+            // only collection types work for agg functions, e.g. SUM(1) & SUM(NULL) return NULL
             val argType = when (val type = nAry.arg.getStaticType()) {
                 is CollectionType -> type.elementType
                 else -> type
             }
             val sourceLocation = nAry.getStartingSourceLocationMeta()
-            return nAry.withStaticType(computeReturnTypeForAggFunc(funcName, argType, sourceLocation))
+            return nAry.withStaticType(computeReturnTypeForAggFunc(funcName.text, argType, sourceLocation))
         }
 
-        fun handleInvalidInputTypeForAggFun(sourceLocation: SourceLocationMeta, aggFunc: SymbolPrimitive, actualType: StaticType, expectedType: StaticType) {
+        private fun handleInvalidInputTypeForAggFun(sourceLocation: SourceLocationMeta, funcName: String, actualType: StaticType, expectedType: StaticType) {
             problemHandler.handleProblem(
                 Problem(
                     sourceLocation = sourceLocation,
                     details = SemanticProblemDetails.InvalidArgumentTypeForFunction(
-                        functionName = aggFunc.text,
+                        functionName = funcName,
                         expectedType = expectedType,
                         actualType = actualType
                     )
                 )
             )
         }
 
-        private fun computeReturnTypeForAggFunc(aggFunc: SymbolPrimitive, argType: StaticType, sourceLocation: SourceLocationMeta): StaticType {
-            // nested type is not supported by agg functions.
-            // i.e. sum(1,2,[3,4]) is not supported
-            val isNullOrMissingOrNumeric: (StaticType) -> Boolean = {
-                it.isSubTypeOf(StaticType.unionOf(StaticType.MISSING, StaticType.NULL, StaticType.NUMERIC))
+        private fun computeReturnTypeForAggFunc(funcName: String, elementType: StaticType, sourceLocation: SourceLocationMeta): StaticType {
+            if (funcName == "count") {
```

Comment:
nit: could keep this within the `when` expression for consistent style

Reply:
I'd like to keep it there so these lines won't get executed: 
```
            val elementTypes = elementType.allTypes

            fun List<StaticType>.convertMissingToNull() = toMutableSet().apply {
                if (contains(StaticType.MISSING)) {
                    remove(StaticType.MISSING)
                    add(StaticType.NULL)
                }
            }

            fun StaticType.isUnknownOrNumeric() = isUnknown() || isNumeric()
```

Reply:
Thanks for explaining. I have no strong opinion here. I will apply this comment.